src/util/string.cpp

```diff
@@ -978,12 +978,14 @@ std::string sanitize_untrusted(std::string_view str, bool keep_escapes)
 			if (i+1 < s.length()) {
 				if (s[i+1] == '(') {
 					if (i+2 < s.length() && check(s[i+2])) {
-						// valid long-form escape sequence
+						// valid long-form escape
 					} else {
 						s.erase(i, 1);
```

Comment:
Is it fine that this is a linear time operation? This makes the stripping itself overall have quadratic time complexity. The string length limit is something like 16k? 65k? which could get heavy quickly if a malicious actor wanted it to (that said, I'm not sure how large our DoS attack surface area is besides this).

It might be a better idea to add everything that's not being stripped to a separate string than to edit the existing string in place, or to rewrite this in such a way that it uses e.g. `std::remove_if`?

Reply:
worst case performance (entire string is escapes):
```
benchmark name                       samples       iterations    est run time
                                     mean          low mean      high mean
                                     std dev       low std dev   high std dev
-------------------------------------------------------------------------------
sanitize_untrusted66000                        100             1 5090218.200000 us 
                                     52863.296440 us 52336.080070 us 53545.146320 us 
                                     3052.340877 us 2425.656089 us 3924.135039 us 
```